src/test/java/com/amazon/opendistroforelasticsearch/security/compliance/ComplianceConfigTest.java

```diff
@@ -0,0 +1,175 @@
+        ComplianceConfig objectUnderTest = (ComplianceConfig) Reflector.getInstance(
+                "com.amazon.opendistroforelasticsearch.security.compliance.ComplianceConfig");
+
+        LoadingCache<String, Set> loadingCache = PowerMockito.mock(LoadingCache.class);
+        HashSet hashSet = new HashSet();
+        hashSet.add("BAZ");
+        Method getMethod = DTUMemberMatcher.method(LoadingCache.class, "get", Object.class);
+        PowerMockito.doReturn(hashSet)
+                .when(loadingCache, getMethod)
+                .withArguments(or(isA(Object.class), isNull(Object.class)));
+        Reflector.setField(objectUnderTest, "cache", loadingCache);
+        Reflector.setField(objectUnderTest, "enabled", true);
+        Reflector.setField(objectUnderTest, "opendistrosecurityIndex", "a/b/c");
+
+        Assert.assertTrue(objectUnderTest.readHistoryEnabledForField("2", "BAZ"));
+    }
+
+    @Test
+    public void testLogDiffsForWrite() throws InvocationTargetException {
+        ComplianceConfig objectUnderTest = (ComplianceConfig) Reflector.getInstance(
+                "com.amazon.opendistroforelasticsearch.security.compliance.ComplianceConfig");
+
+        Reflector.setField(objectUnderTest, "logDiffsForWrite", false);
+        Reflector.setField(objectUnderTest, "logWriteMetadataOnly", false);
+
+        Assert.assertFalse(objectUnderTest.logDiffsForWrite());
+    }
+
+
+    @Test
+    public void testLogReadMetadataOnly() throws InvocationTargetException {
+        final ComplianceConfig objectUnderTest = (ComplianceConfig) Reflector.getInstance(
+                "com.amazon.opendistroforelasticsearch.security.compliance.ComplianceConfig");
+        Reflector.setField(objectUnderTest, "logReadMetadataOnly", false);
+
+        Assert.assertFalse(objectUnderTest.logReadMetadataOnly());
+    }
+
+    @Test
+    public void testIsIndexImmutable() throws Exception {
+        ComplianceConfig objectUnderTest = (ComplianceConfig) Reflector.getInstance(
+                "com.amazon.opendistroforelasticsearch.security.compliance.ComplianceConfig");
+
+        Assert.assertFalse(objectUnderTest.isIndexImmutable("foo"));
+
+        Set<String> hashset = new HashSet<>();
+        Reflector.setField(objectUnderTest, "immutableIndicesPatterns", hashset);
+        Reflector.setField(objectUnderTest, "enabled", true);
+        Reflector.setField(objectUnderTest, "opendistrosecurityIndex", "a/b/c");
+
+        Assert.assertFalse(objectUnderTest.isIndexImmutable("foo"));
+    }
+
+    @Test
+    public void testGetSalt16() throws InvocationTargetException {
+        ComplianceConfig objectUnderTest = (ComplianceConfig) Reflector.getInstance(
+                "com.amazon.opendistroforelasticsearch.security.compliance.ComplianceConfig");
+        Reflector.setField(objectUnderTest, "salt16", new byte[]{});
+
+        Assert.assertArrayEquals(new byte[]{}, objectUnderTest.getSalt16());
```

Comment:
This test just assigns a field reflectively and tests the getter

Reply:
Removed test case